File: Python/求最大公因数和最小公倍数.py
import sys
a=[]
for i in range(1,3):
	a.append(int(sys.argv[i]))
m,n=a[0],a[1]
with open('result.txt','w') as fp:
	if m<1 or n<1 :
		fp.write('I')
	else :
		if m>=n :
			a,b=m,n
		else :
			a,b=n,m
		while b>0 :
			c=a%b
			a,b=b,c
		fp.write(str(a)+' '+str(int(m*n/a)))

File: C/求最大公因数和最小公倍数.c
#include<stdio.h>
#include<stdlib.h>
int main(int argc,char** argv)
{
	FILE *fp;
	if((fp=fopen("result.txt","w"))==NULL)
		return -1;
	long m,n,a,b,c,i,x[3]={0};
	for(i=1;i<3;i++)
		x[i]=atoi(argv[i]);
	m=x[1];
	n=x[2];
	if(m<1||n<1) fprintf(fp,"I");
	else
	{
		if(m>=n)
		{
			a=m;
			b=n;
		}
		else
		{
			a=n;
			b=m;
		}
		while(b>0)
		{
			c=a%b;
			a=b;
			b=c;
		}
		fprintf(fp,"%ld %ld",a,m*n/a);
	}
	fclose(fp);	
	return 0;
}
